binary-decode.go

```diff
@@ -648,6 +652,7 @@ func (cdc *Codec) decodeReflectBinaryStruct(bz []byte, info *TypeInfo, rv reflec
 	default:
 		// Read each field.
 		for _, field := range info.Fields {
+			lastFieldNum++
```

Comment:
It's semantically awkward to increment it here in the beginning...

Reply:
See [below](https://github.com/tendermint/go-amino/pull/155/files#r191466725). I don't increment now but set `lastFieldNum = fieldnum` instead (not at the end but right after the check)